### DISCUSSION TITLE
Can HuggingFace's utilities for generation be used with llama.cpp?

### DISCUSSION THREAD

**Discussion by chashimo** [ggml-org/llama.cpp#4139] [Q&A] [answered]
Hello, llama.cpp developers.

I would like to use HuggingFace's [utilities for generation ](https://huggingface.co/docs/transformers/internal/generation_utils) with llama.cpp. Can I use all of their utilities such as SequenceBiasLogitsProcessor? Thanks in advance!

**Comment by KerfuffleV2** [comment, marked as answer]
I think the short answer here is "no". That's meant for interfacing with their PyTorch-based stuff.

It's not impossible that at some point you might be able to use it (someone would have to write some kind of interface, which wouldn't be a simple matter), but as far as I know there's currently no way. If I had to guess, I'd say your best bet is to wait and see if they do something to make it work with their Candle project which I believe can load GGUF files.

**Reply by chashimo** [reply]
Thanks, @KerfuffleV2! I will wait.
